variables.tf

```diff
@@ -122,12 +128,23 @@ variable "glacier_transition_days" {
   description = "Number of days after which to move the data to the glacier storage tier"
 }
 
+variable "deeparchive_transition_days" {
+  type        = number
+  default     = 90
+  description = "Number of days after which to move the data to the glacier deep archive storage tier"
+}
+
 variable "enable_glacier_transition" {
   type        = bool
-  default     = true
+  default     = false
```

Comment:
@bcoddens why to change default behaviour?

Reply:
You are correct,just tested it.
When you have in variables.tf:

variable "enable_glacier_transition" {
  type        = bool
  default     = true
  description = "Enables the transition to AWS Glacier which can cause unnecessary costs for huge amount of small files"
}

== The default

variable "enable_deeparchive_transition" {
  type        = bool
  default     = false
  description = "Enables the transition to AWS Glacier Deep Archive which can cause unnecessary costs for huge amount of small files"
}

And this in your main.tf:

  enable_glacier_transition = "false"
  enable_deeparchive_transition              = "true"

It works fine according to the plan output:

          + noncurrent_version_expiration {
              + days = 90
            }

          + noncurrent_version_transition {
              + days          = 60
              + storage_class = "DEEP_ARCHIVE"
            }

          + transition {
              + days          = 90
              + storage_class = "DEEP_ARCHIVE"
            }
        }